extensions/amp-ad-network-adzerk-impl/0.1/amp-ad-network-adzerk-impl.js

```diff
@@ -0,0 +1,272 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  AmpA4A,
+  NO_CONTENT_RESPONSE,
+  CreativeMetaDataDef,
+} from '../../amp-a4a/0.1/amp-a4a';
+import {parse as mustacheParse, render as mustacheRender,
+  setUnescapedSanitizier} from '../../../third_party/mustache/mustache';
```

Comment:
why did we decide not to use template service?

it's helpful to rely on the same code entrance so we can fix any potential security hole in one place.

Reply:
Honestly, I didn't see it until I have 80% implemented what you see here :) . Will change to use template service.

Reply:
Are we expecting to see templates other than of type "amp-mustache"? If not, why not just use the AmpMustache extension?

Reply:
Further, I'm not sure what it means to use the template service here. I've been fiddling with it for a while now, and it seems like I'd need to provide my own instance of TemplateImpl, right? If so, I'd need to write the render method, which would need to call the mustache functions. Or am I missing something?

Reply:
you can take a look at the usage of templateservice in amp-ad-custom.js

Reply:
Seems like the element passed to `findAndRenderTemplate` needs to be attached to the main doc. Otherwise [this value](https://github.com/ampproject/amphtml/blob/master/src/service/template-impl.js#L53) comes out null, which causes a failure [here in the AmpMustache extension](https://github.com/ampproject/amphtml/blob/master/extensions/amp-mustache/0.1/amp-mustache.js#L45). Is there any way of rendering a template on a non-attached element?